main.js

```diff
@@ -0,0 +1,334 @@
+
+  const fetchedData = await fetch(url, {
+    method: "Get",
+    dataType: "json",
+  });
+
+  const nowData = await fetchedData.json();
+
+  return nowData;
+}
+
+async function addNow(data) {
+  const nowSection = document.querySelector("#now");
+  nowSection.replaceChildren();
+
+  const template = `      
+    <div class="row justify-content-center forecast-today my-3" id="now">
+      <div class="col-md-6">
+        <h3 class="text-center">${data.main.temp.toFixed(1)}°</h3>
+        <h4 class="text-center">${data.name}</h4>
+        <h5 class="text-center">${data.weather[0].main}</h5>
+      </div>
+      <div class="col-md-6">
+        <img
+          class="img-fluid mx-auto d-block"
+          src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png" )
+          alt="unable to load image"
+        />
+      </div>
+    </div>
+  `;
+
+  nowSection.insertAdjacentHTML("beforeend", template);
+}
+
+async function getFiveDay() {
+  try {
+    const coordinates = await fetchCoordinates();
+    const fiveDayData = await fetchFiveDayData(coordinates);
+    const processedData = await processFiveDayData(fiveDayData);
+    addFiveDay(processedData);
+  } catch (error) {
+    console.error("Error processing data:", error);
+  }
+}
+
+async function fetchCoordinates() {
+  const query = document.querySelector("#query").value.replace(/\s+/g, "%20");
+
+  const coordinatesCall = await fetch(
+    `https://api.openweathermap.org/geo/1.0/direct?q=${query}&limit=1&appid=${apiKey}
+    `,
+    {
+      method: "GET",
+      dataType: "json",
+    }
+  );
+
+  const coordinatesArray = await coordinatesCall.json();
+  const coordinates = coordinatesArray[0];
+
+  return coordinates;
+}
+
+async function fetchFiveDayData(coordinates) {
+  const fiveDayURL = `https://api.openweathermap.org/data/2.5/forecast?lat=${coordinates.lat}&lon=${coordinates.lon}&appid=${apiKey}&units=${units}`;
+
+  const fetchFiveDay = await fetch(fiveDayURL, {
+    method: "GET",
+    dataType: "json",
+  });
+
+  const fiveDayData = await fetchFiveDay.json();
+
+  return fiveDayData;
+}
+
+async function processFiveDayData(data) {
+  // create an array for each of the five days
+
+  const dividedDayData = [];
+
+  // process information in each array to produce the correct temperature estimate and the most commonly listed icon code and the name of the day. This information could be stored in an object that summarizes the information. This would make it easier to reference in the html tempalate.
+
+  // update html template with each day's information
+
+  function processDayData(day) {
+    const dayData = {
+      avgTemp: 0,
+      iconSummary: null,
+      weatherSummary: null,
+      dayName: "",
+    };
+
+    let tempAcc = 0;
+    let icons = {};
+    let weatherDescriptions = {};
+
+    // identify the appropriate data range
+    let indexRange = [];
+
+    if (day === 1) {
```

Comment:
maybe just make this a dictionary lookup

Reply:
Yeah, you could store this mapping in an object like this:

```ts
const ranges = {
  1: [0, 7],
  2: [0, 15],
  ...etc
}

const indexRange = ranges[day];
```

However, lines `168-184` could be replaced with one line with simple math...

I'll hold off on giving you the solution... but I have it :D

Reply:
Figured this out!